tools/version_docomentaion.py

```diff
@@ -0,0 +1,84 @@
+import subprocess
+import argparse
+import logging
+import shutil
+import os
+import re
+
+logging.basicConfig(format='%(asctime)s %(message)s')
+logging.getLogger().setLevel(logging.INFO)
+
+REPO_URL = "https://github.com/openshift/assisted-service.git"
+valid_commit_regex = '^([A-Z]+-[0-9]+|#[0-9]+|merge|no-issue)'
+
+############################################################################################
+# This is used to generate the documentation of assisted-service between different versions.
+############################################################################################
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--from-version", help="From version to document", type=str)
+parser.add_argument("--to-version", help="To version to document", type=str)
+parser.add_argument("--documentation-dir", help="deployment yaml file to update", type=str,
+                    default=os.path.join(os.path.dirname(__file__), "../versions_documentation"))
+args = parser.parse_args()
+
+
+def main():
+    version_documentation_list = list()
+
+    if not os.path.exists(args.documentation_dir):
+        os.makedirs(args.documentation_dir)
+
+    os.mkdir("temp")
```

Comment:
why do you need this temp dir?

Reply:
we clone the repo to it to get the logs for processing